src/parser.rs

```diff
@@ -748,36 +748,26 @@ impl<'tokens, 'src> Parser<'tokens, 'src> {
     let name = Name::from_identifier(self.presume(Identifier)?);
     let lexeme = name.lexeme();
 
-    if Keyword::AllowDuplicateRecipes == lexeme {
-      let value = self.parse_set_bool()?;
-      return Ok(Set {
-        value: Setting::AllowDuplicateRecipes(value),
-        name,
-      });
+    let set_bool: Option<Setting> = if Keyword::AllowDuplicateRecipes == lexeme {
```

Comment:
This is a nice refactor, imo, but I think you could take it one more step if you wanted and turn this into a `match` expression instead of a series of `if/else`.

Just looking around out of curiosity and going to throw my 2 cents in (feel free to take it or leave it).

Reply:
I originally thought it wasn't possible to do this because `lexeme` is `&str` and can't be directly pattern-matched against `Keyword`, but I think if I parse the `lexeme` first I can pattern-match against that and make this bit of code a bit more concise. Thanks for the suggestion!